drivers/storage/portworx/component/security.go

```diff
@@ -0,0 +1,212 @@
+	SecuritySystemSecretKey = "system-secret"
+)
+
+type security struct {
+	k8sClient client.Client
+}
+
+// Initialize initializes the componenet
+func (c *security) Initialize(
+	k8sClient client.Client,
+	k8sVersion version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	c.k8sClient = k8sClient
+}
+
+// IsEnabled checks if the components needs to be enabled based on the StorageCluster
+func (c *security) IsEnabled(cluster *corev1alpha1.StorageCluster) bool {
+	return pxutil.SecurityEnabled(cluster)
+}
+
+// Reconcile reconciles the component to match the current state of the StorageCluster
+func (c *security) Reconcile(cluster *corev1alpha1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	err := c.createPrivateKeysSecret(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete deletes the component if present
+func (c *security) Delete(cluster *corev1alpha1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	err := c.deletePrivateKeysSecret(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MarkDeleted marks the component as deleted in situations like StorageCluster deletion
+func (c *security) MarkDeleted() {
+
+}
+
+func (c *security) getPrivateKeyOrGenerate(cluster *corev1alpha1.StorageCluster, envVarKey, secretName, secretKey string) (string, error) {
+	var privateKey string
+	var err error
+
+	// check for pre-configured shared secret
+	for _, envVar := range cluster.Spec.Env {
+		if envVar.Name == envVarKey {
+			privateKey, err = pxutil.GetValueFromEnvVar(context.TODO(), c.k8sClient, &envVar, cluster.Namespace)
+			if err != nil {
+				return "", err
+			}
+
+			return privateKey, nil
+		}
+	}
+
+	// If we did not find a secret, generate and add it
+	if privateKey == "" {
+		privateKey, err = generateAuthSecret()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// Add to cluster env if we had to generate it
+	err = addPrivateKeyEnv(cluster, envVarKey, secretName, secretKey)
+	if err != nil {
+		return "", err
+	}
+
+	return privateKey, nil
+}
+
+func (c *security) createPrivateKeysSecret(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	var sharedSecretKey, internalSystemSecretKey string
+	var err error
+
+	sharedSecretKey, err = c.getPrivateKeyOrGenerate(
+		cluster,
+		pxutil.EnvKeyPortworxAuthJwtSharedSecret,
+		SecurityPXPrivateKeysSecretName,
+		SecuritySharedSecretKey,
+	)
+	if err != nil {
+		return err
+	}
+
+	internalSystemSecretKey, err = c.getPrivateKeyOrGenerate(
+		cluster,
+		pxutil.EnvKeyPortworxAuthSystemKey,
+		SecurityPXPrivateKeysSecretName,
+		SecuritySystemSecretKey,
+	)
+	if err != nil {
+		return err
+	}
+
+	privateKeysSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      SecurityPXPrivateKeysSecretName,
+			Namespace: cluster.Namespace,
```

Comment:
Any reason we are not adding owner reference here? 
Is the intention to not delete the secret when storage cluster is deleted?